WeakAuras/GenericTrigger.lua

```diff
@@ -5126,6 +5126,57 @@ Private.GetCurrencyInfoForTrigger = function(trigger)
   end
 end
 
+Private.ExecEnv.GetCurrencyAccountInfo = function(currencyId)
+  local currencyInfo = C_CurrencyInfo.GetCurrencyInfo(currencyId)
+
+  -- Gather currency data for account characters
+  if WeakAuras.IsRetail() and currencyInfo and currencyInfo.isAccountTransferable then
+    if C_CurrencyInfo.IsAccountCharacterCurrencyDataReady() then
+      currencyInfo.accountQuantity = currencyInfo.quantity
+      currencyInfo.realAccountQuantity = currencyInfo.quantity
+      if currencyInfo.transferPercentage then
+          currencyInfo.realCharacterQuantity = math.floor(currencyInfo.quantity * (currencyInfo.transferPercentage/100))
+      end
+
+      currencyInfo.accountCurrencyData = C_CurrencyInfo.FetchCurrencyDataFromAccountCharacters(currencyId) or {}
+
+      for _, currencyData in ipairs(currencyInfo.accountCurrencyData) do
+        if currencyData.quantity then
+          local realQuantity = currencyData.quantity
+          if currencyInfo.transferPercentage and currencyInfo.transferPercentage > 0 and currencyInfo.transferPercentage < 100 then
+            realQuantity = math.floor(realQuantity * (currencyInfo.transferPercentage/100))
+          end
+          currencyData.realCharacterQuantity = realQuantity
```

Comment:
This setting of realCharacterQuantity for each accountCurrencyData looks wrong.

Reply:
The logic there was correct, but also contained unneeded sanity checks. I've simplified this bit.